editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/properties/AdvancedBeanPropertiesSection.java

```diff
@@ -0,0 +1,391 @@
+						/*
+						 * (non-Javadoc)
+						 * @see org.eclipse.core.databinding.validation.IValidator#validate(java.lang.Object)
+						 */
+						@Override
+						public IStatus validate(Object value) {
+							if (value != null && value instanceof String && value.toString().trim().length()>0) {
+								return ValidationStatus.ok();
+							}
+							return ValidationStatus.error("Parameter " + prop.getName() + " is a mandatory field and cannot be empty.");
+						}
+					};
+				}
+				
+				// UNSUPPORTED PROPERTIES / REFS
+            } else if (CamelComponentUtils.isUnsupportedProperty(prop)) {
+				new UnsupportedParameterPropertyUICreatorForAdvanced(dbc, modelMap, eip, selectedEP, p, page, getWidgetFactory()).create();
+            }
+
+			if (uiObservable != null) {
+				// create UpdateValueStrategy and assign to the binding
+				UpdateValueStrategy strategy = new UpdateValueStrategy();
+				strategy.setBeforeSetValidator(validator);
+
+				// create observables for the Map entries
+				modelObservable = Observables.observeMapEntry(modelMap, p.getName());
+				// bind the observables
+				Binding bindValue = dbc.bindValue(uiObservable, modelObservable, strategy, null);
+				ControlDecorationSupport.create(bindValue, SWT.TOP | SWT.LEFT);
+			}
+        }
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see
+     * org.eclipse.ui.views.properties.tabbed.AbstractPropertySection#createControls
+     * (org.eclipse.swt.widgets.Composite,
+     * org.eclipse.ui.views.properties.tabbed.TabbedPropertySheetPage)
+     */
+    @Override
+    public void createControls(Composite parent, TabbedPropertySheetPage aTabbedPropertySheetPage) {
+        this.toolkit = new FormToolkit(parent.getDisplay());
+        super.createControls(parent, aTabbedPropertySheetPage);
+
+        // now setup the file binding properties page
+        parent.setLayout(new GridLayout());
+        parent.setLayoutData(new GridData(GridData.FILL_BOTH));
+
+        form = toolkit.createForm(parent);
+        form.setLayoutData(new GridData(GridData.FILL_BOTH));
+        form.getBody().setLayout(new GridLayout(1, false));
+
+        Composite sbody = form.getBody();
+
+        tabFolder = new CTabFolder(sbody, SWT.TOP | SWT.FLAT);
+        toolkit.adapt(tabFolder, true, true);
+        tabFolder.setLayoutData(new GridData(GridData.FILL_BOTH));
+
+        Color selectedColor = toolkit.getColors().getColor(IFormColors.SEPARATOR);
+        tabFolder.setSelectionBackground(new Color[] { selectedColor, toolkit.getColors().getBackground() }, new int[] { 20 }, true);
+        tabFolder.setCursor(FormsResources.getHandCursor());
+        toolkit.paintBordersFor(tabFolder);
+
+        form.setText("Advanced Properties");
+        toolkit.decorateFormHeading(form);
+        
+        form.layout();
+        tabFolder.setSelection(0);
+    }
+
+    class AttributeTextParameterPropertyModifyListenerForAdvanced extends AbstractTextParameterPropertyModifyListener {
```

Comment:
would be better to extract the class in its own file would be also easier to test it